homeassistant/components/smhi/weather.py

```diff
@@ -220,13 +238,17 @@ def forecast(self) -> list[Forecast] | None:
                 {
                     ATTR_FORECAST_TIME: forecast.valid_time.isoformat(),
                     ATTR_FORECAST_NATIVE_TEMP: forecast.temperature_max,
-                    ATTR_FORECAST_NATIVE_TEMP_LOW: forecast.temperature_min,
                     ATTR_FORECAST_NATIVE_PRECIPITATION: forecast.total_precipitation,
                     ATTR_FORECAST_CONDITION: condition,
                     ATTR_FORECAST_NATIVE_PRESSURE: forecast.pressure,
                     ATTR_FORECAST_WIND_BEARING: forecast.wind_direction,
                     ATTR_FORECAST_NATIVE_WIND_SPEED: forecast.wind_speed,
                 }
+                | (
+                    {ATTR_FORECAST_NATIVE_TEMP_LOW: forecast.temperature_min}
+                    if not self._hourly
+                    else {}
+                )
```

Comment:
Please put this in a separate update to the dict instead of this complex looking expression.